### PR TITLE
[SW-1489] Use TypeConverters on AutoML pysparkling wrapper

### PR DESCRIPTION
**Pull request by jakubhava**
Also fixes bug in H2OAutoML constructor. `ClassSamplingFactors` was not converted to double values

### CODE REVIEW

File: py/ai/h2o/sparkling/ml/params/H2OCommonParams.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
 
 from h2o.utils.typechecks import assert_is_type
 from pyspark.ml.param import *
+from pyspark.ml.param import TypeConverters
```
Comment (mn-mikke) [inline py/ai/h2o/sparkling/ml/params/H2OCommonParams.py]: Isn't this line extra if we have the line above?

File: py/ai/h2o/sparkling/ml/algos/H2OAutoML.py [hunk 2]
```diff
@@ -64,18 +63,9 @@ def setParams(self, featuresCols=[], labelCol="label", allStringColumnsToCategor
 
         kwargs = get_input_kwargs(self)
 
-        validateEnumValues(self._H2OAutoMLParams__getAutomlAlgoEnum(), kwargs, "includeAlgos", nullEnabled=True)
-        validateEnumValues(self._H2OAutoMLParams__getAutomlAlgoEnum(), kwargs, "excludeAlgos", nullEnabled=True)
-        validateEnumValue(self._H2OAutoMLParams__getStoppingMetricEnum(), kwargs, "stoppingMetric")
-        validateEnumValue(self._H2OAutoMLParams__getSortMetricEnum(), kwargs, "sortMetric")
-
         if "projectName" in kwargs and kwargs["projectName"] is None:
```
Comment (mn-mikke) [inline py/ai/h2o/sparkling/ml/algos/H2OAutoML.py]: Shouldn't this be moved to parama spec as well?

Comment (jakubhava) [reply py/ai/h2o/sparkling/ml/algos/H2OAutoML.py]: This is a bit special -> we are generating random string on Scala part, but that is not propagated to Python. But please see https://github.com/h2oai/sparkling-water/pull/1419 where I should improve this part

File: py/ai/h2o/sparkling/ml/params/H2OCommonParams.py (hunk 1 above)
Comment (jakubhava) [reply py/ai/h2o/sparkling/ml/params/H2OCommonParams.py]: Good catch, fixed

File: py/ai/h2o/sparkling/ml/algos/H2OAutoML.py (hunk 2 above)
Comment (mn-mikke) [reply py/ai/h2o/sparkling/ml/algos/H2OAutoML.py]: After #1419 gets merged, should this be completely removed? Just wondering whether the constructor parameter and the setter will behave the same way.

Comment (jakubhava) [reply py/ai/h2o/sparkling/ml/algos/H2OAutoML.py]: Yup, my bad #1419 should delete that, will fix it on that PR side

**Pull request merged**